src/EnergyPlus/DataGlobalConstants.cc

```diff
@@ -652,6 +658,146 @@ namespace DataGlobalConstants {
+        //       MODIFIED       na
+        //       RE-ENGINEERED  na
+
+        // PURPOSE OF THIS FUNCTION:
+        // Validates fuel types and sets output strings
+
+        auto const SELECT_CASE_var(FuelTypeInput);
+
+        if (SELECT_CASE_var == "ELECTRICITY") {
+            FuelType = "Electric";
+
+        } else if (SELECT_CASE_var == "NATURALGAS") {
+            FuelType = "Gas";
+
+        } else if (SELECT_CASE_var == "DIESEL") {
+            FuelType = "Diesel";
+
+        } else if (SELECT_CASE_var == "GASOLINE") {
+            FuelType = "Gasoline";
+
+        } else if (SELECT_CASE_var == "COAL") {
+            FuelType = "Coal";
+
+        } else if (SELECT_CASE_var == "FUELOILNO1") {
+            FuelType = "FuelOil#1";
+
+        } else if (SELECT_CASE_var == "FUELOILNO2") {
+            FuelType = "FuelOil#2";
+
+        } else if (SELECT_CASE_var == "PROPANE") {
+            FuelType = "Propane";
+
+        } else if (SELECT_CASE_var == "OTHERFUEL1") {
+            FuelType = "OtherFuel1";
+
+        } else if (SELECT_CASE_var == "OTHERFUEL2") {
+            FuelType = "OtherFuel2";
+
+        } else {
+            FuelTypeErrorsFound = true;
+        }
+    }
+
+    void ValidateFuelTypeWithFuelTypeNum(std::string FuelTypeInput)
+    {
+        // FUNCTION INFORMATION:
+        //       AUTHOR         Dareum Nam
+        //       DATE WRITTEN   May 2020
+        //       MODIFIED       na
+        //       RE-ENGINEERED  na
+
+        // PURPOSE OF THIS FUNCTION:
+        // Validates fuel types and sets output strings with fuel type number (DXCoils.cc and HVACVariableRefrigerantFlow.cc)
+
+        if (UtilityRoutines::SameString(FuelTypeInput, "Electricity")) {
+            FuelTypeNum = 1; // FuelTypeElectricity
+        } else if (UtilityRoutines::SameString(FuelTypeInput, "NaturalGas")) {
+            FuelTypeNum = 2; // FuelTypeNaturalGas
+        } else if (UtilityRoutines::SameString(FuelTypeInput, "Propane")) {
+            FuelTypeNum = 3; // FuelTypePropaneGas
+        } else if (UtilityRoutines::SameString(FuelTypeInput, "Diesel")) {
+            FuelTypeNum = 4; // FuelTypeDiesel
+        } else if (UtilityRoutines::SameString(FuelTypeInput, "Gasoline")) {
+            FuelTypeNum = 5; // FuelTypeGasoline
+        } else if (UtilityRoutines::SameString(FuelTypeInput, "FuelOilNo1")) {
+            FuelTypeNum = 6; // FuelTypeFuelOil1
+        } else if (UtilityRoutines::SameString(FuelTypeInput, "FuelOilNo2")) {
+            FuelTypeNum = 7; // FuelTypeFuelOil2
+        } else if (UtilityRoutines::SameString(FuelTypeInput, "OtherFuel1")) {
+            FuelTypeNum = 8; // FuelTypeOtherFuel1
+        } else if (UtilityRoutines::SameString(FuelTypeInput, "OtherFuel2")) {
+            FuelTypeNum = 9; // FuelTypeOtherFuel2
+        } else {
+            FuelTypeErrorsFound = true;
+        }
+    }
+
+    void ValidateFuelTypeWithAssignResourceTypeNum(std::string FuelTypeInput)
```

Comment:
As I mentioned in another comment, you need to pass in and out the `FuelType` string and the `FuelTypeNum` integer.  This input string also needs to be a constant reference so that it isn't copied when this function is called.  The new signature should be:

```
bool ValidateFuelTypeWithAssignResourceTypeNum(
    std::string const & FuelTypeInput,
    std::string & FuelType,
    int & FuelTypeNum
);
```